lib/improver/tests/utilities/test_TemporalInterpolation.py

```diff
@@ -158,77 +158,92 @@ def test_time_list_out_of_bounds(self):
+        self.assertEqual(result.coord('forecast_reference_time'),
+                         self.cube.coord('forecast_reference_time'))
+        self.assertEqual(
+            str(result.coord('forecast_reference_time').points.dtype), 'int64')
+        # Check forecast_period.
         self.assertEqual(result.coord('forecast_period'),
                          self.cube.coord('forecast_period'))
         self.assertEqual(str(result.coord('forecast_period').points.dtype),
                          'int32')
-        self.assertEqual(result.coord('extra_coord'),
-                         self.cube.coord('extra_coord'))
-        self.assertEqual(str(result.coord('extra_coord').points.dtype),
-                         'float32')
 
-    def test_time_types(self):
-        """Test that only the coords associated with time have been updated."""
+    def test_check_bounds(self):
+        """Test that a cube is returned.with the right types"""
         plugin = TemporalInterpolation(interval_in_minutes=60)
-        result = plugin.check_cube_coords_dtype(self.cube,
-                                                self.bad_coords,
-                                                dim_coords=['time'])
-        # Time and forecast_period reverted back to original type
+        cube = self.cube
+        # Use of guess_bounds converts datatype to float64.
+        cube.coord("time").guess_bounds()
+        cube.coord("forecast_period").guess_bounds()
+
+        result = plugin.enforce_time_coords_dtype(cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        # All coordinates including bounds converted to the
+        # desired units and datatypes.
+        # Check time coordinate.
         self.assertEqual(result.coord('time'),
                          self.cube.coord('time'))
         self.assertEqual(str(result.coord('time').points.dtype),
                          'int64')
+        self.assertEqual(str(result.coord('time').bounds.dtype),
+                         'int64')
+        # Check forecast_reference_time coordinate.
+        self.assertEqual(result.coord('forecast_reference_time'),
+                         self.cube.coord('forecast_reference_time'))
+        self.assertEqual(
+            str(result.coord('forecast_reference_time').points.dtype), 'int64')
+        # Check forecast_period coordinate.
         self.assertEqual(result.coord('forecast_period'),
                          self.cube.coord('forecast_period'))
         self.assertEqual(str(result.coord('forecast_period').points.dtype),
                          'int32')
-        # Extra coordinate left as changed.
-        self.assertEqual(result.coord('extra_coord'),
-                         self.bad_coords.coord('extra_coord'))
-        self.assertEqual(str(result.coord('extra_coord').points.dtype),
-                         'float64')
+        self.assertEqual(str(result.coord('forecast_period').bounds.dtype),
+                         'int32')
```

Comment:
As above: you need to make some copies to check datatypes.

Can we also please have a test / some `assert` statements for the unit conversion.

Reply:
Added.